src/Microsoft.Health.Dicom.Core/Features/Store/StoreService.cs

```diff
@@ -262,14 +262,14 @@ private void LogFailure(int index, DicomDataset dicomDataset, StoreValidationRes
     private void DropInvalidMetadata(StoreValidationResult storeValidatorResult, DicomDataset dicomDataset, Partition partition)
     {
         var identifier = dicomDataset.ToInstanceIdentifier(partition);
-        foreach (DicomTag tag in storeValidatorResult.InvalidTagErrors.Keys)
+        foreach (KeyValuePair<DicomTag, StoreErrorResult> error in storeValidatorResult.InvalidTagErrors)
```

Comment:
You should be able to use a tuple here:
```suggestion
        foreach ((DicomTag tag, StoreErrorResult result) in storeValidatorResult.InvalidTagErrors)
```